packages/env/lib/commands/start.js

```diff
@@ -141,23 +139,10 @@ module.exports = async function start( { spinner, debug, update, xdebug } ) {
 			: [],
 	} );
 
-	await Promise.all( [
-		makeConfigWritable( 'development', config ),
-		makeConfigWritable( 'tests', config ),
-	] );
-
 	// Only run WordPress install/configuration when config has changed.
 	if ( shouldConfigureWp ) {
 		spinner.text = 'Configuring WordPress.';
 
-		if ( config.coreSource === null ) {
```

Comment:
From reading the code and experimenting, it seems to me that `config.coreSource` is always undefined, and this code never gets executed. I tested with `core: null`, and `core: WordPress/WordPress` in `.wp-env.json`, and filesystem permissions were always fine (this conditional claims to solve filesystem permissions).

It seems that the issues this conditional was meant to resolve have been resolved in the Docker Hub wordpress image:  https://github.com/docker-library/wordpress/issues/436

Reply:
Cool. I think `config.coreSource` is used in `buildDockerComposeConfig` to map over local or git/remote WordPress sources. But if that issue is resolved, it's great that we can remove it. I believe this conditional was added here: https://github.com/WordPress/gutenberg/pull/20406.

cc @noisysocks, do you know how to test the problem this was solving?

Reply:
Note that `config.coreSource` in `buildDockerComposeConfig` is not actually `config.coreSource` but `config.env.development.coreSource` or `config.env.tests.coreSource`

Reply:
Actually that's nonsense, sorry, it is actually `config.coreSource`

Reply:
No this is all completely wrong, I'll update it

Reply:
I fixed the incorrect usage of `config.coreSource` in `buildDockerComposeConfig`, added unit tests for it, and tested manually that everything still works fine